api/src/indigo_structure_checker.cpp

```diff
@@ -0,0 +1,205 @@
+/****************************************************************************
+ * Copyright (C) from 2009 to Present EPAM Systems.
+ *
+ * This file is part of Indigo toolkit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+
+#include "indigo_structure_checker.h"
+#include "api/indigo.h"
+#include "api/src/indigo_molecule.h"
+#include "api/src/indigo_reaction.h"
+#include "base_cpp/tlscont.h"
+#include "molecule/molecule_automorphism_search.h"
+#include "third_party/rapidjson/stringbuffer.h"
+#include "third_party/rapidjson/writer.h"
+#include <functional>
+#include <regex>
+#include <string>
+
+using namespace indigo;
+
+struct CheckParams
+{
+    int check_flags = StructureChecker2::CHECK_ALL;
+    std::vector<int> selected_atoms;
+    std::vector<int> selected_bonds;
+};
+
+static constexpr const char* checkTypeName[] = {
+    IndigoStructureChecker::CHECK_NONE_TXT,      IndigoStructureChecker::CHECK_LOAD_TXT,         IndigoStructureChecker::CHECK_VALENCE_TXT,
+    IndigoStructureChecker::CHECK_RADICAL_TXT,   IndigoStructureChecker::CHECK_PSEUDOATOM_TXT,   IndigoStructureChecker::CHECK_STEREO_TXT,
+    IndigoStructureChecker::CHECK_QUERY_TXT,     IndigoStructureChecker::CHECK_OVERLAP_ATOM_TXT, IndigoStructureChecker::CHECK_OVERLAP_BOND_TXT,
+    IndigoStructureChecker::CHECK_RGROUP_TXT,    IndigoStructureChecker::CHECK_SGROUP_TXT,       IndigoStructureChecker::CHECK_TGROUP_TXT,
+    IndigoStructureChecker::CHECK_CHIRALITY_TXT, IndigoStructureChecker::CHECK_CHIRAL_FLAG_TXT,  IndigoStructureChecker::CHECK_3D_COORD_TXT,
+    IndigoStructureChecker::CHECK_CHARGE_TXT,    IndigoStructureChecker::CHECK_SALT_TXT,         IndigoStructureChecker::CHECK_AMBIGUOUS_H_TXT,
+    IndigoStructureChecker::CHECK_COORD_TXT,     IndigoStructureChecker::CHECK_V3000_TXT,        IndigoStructureChecker::CHECK_ALL_TXT};
+
+static CheckParams check_params_from_string(const char* params)
+{
+    CheckParams r;
```

Comment:
It is worth to avoid one letter variable names

Reply:
When I worked in embedded software development (see MStar inc. in my CV), it was a strict coding style requirement to have exactly one return statement at the end of function.
E.g. let you have to add a finalization code, free resources etc. You add the code at the end of func..... oops! it has 100500 returns from its body and you even can't be sure that you've found all of them.

I am used to following this style.
In my code "var r" always means return value like "var i" means iterator.

I write

bool r;
if (smth)
  r = true;
else
 r = false
return r;

instead of

if (smth)
  return true;
else
 return false

Of course, it's possible to change "r" to e.g. "ret". But it appears in many places and this requires significant work.